src/pages/groups/List.js

```diff
@@ -151,11 +151,14 @@ export const GroupList = ajaxCaller(class GroupList extends Component {
 
     try {
       this.setState({ isDataLoaded: false, creatingNewGroup: false, deletingGroup: false, updating: false })
-      const groups = await Groups.list()
-      this.setState({
-        isDataLoaded: true,
-        groups: _.sortBy('group.groupName', groups)
-      })
+      const rawGroups = await Groups.list()
+      const groupRoles = _.reduce((c, g) => { (c[g.groupName] || (c[g.groupName] = [])).push(g.role); return c }, {}, rawGroups)
+      const groups = _.flow(
+        _.uniqBy('groupName'),
+        _.map(group => _.set('role', groupRoles[group.groupName], group)),
+        _.sortBy('groupName')
+      )(rawGroups)
```

Comment:
Lodash tip: There's a `groupBy` function that would be a good fit here (and probably above as well). It turns a list of items into an object of arrays of items, grouped by whatever you key want. So this could be something like:
```
_.flow(
  _.groupBy('groupName'),
  _.map(gs => ({ ...gs[0], role: _.map('role', gs) })),
  _.sortBy('groupName')
)(rawGroups)

Reply:
Thanks. I tried to find a way to use `groupBy` but couldn't find quite the right transform. I think my mind couldn't get past the need for `gs[0]`, though using `uniqBy` is effectively the same thing. I do like how your suggestion avoids the separate computation of `groupRoles` so I'll make the change.